### PR TITLE
Add `skipn_skipn`

### PR DESCRIPTION
**Pull request by stepchowfun**
The standard library has a lemma called `firstn_firstn`, but no analogous lemma for `skipn`. I found myself needing this little fact in one of my recent developments: `skipn x (skipn y l) = skipn (x + y) l`.

### CODE REVIEW

File: doc/changelog/11-standard-library/16632-patch-1.rst
Comment (Alizter) [inline doc/changelog/11-standard-library/16632-patch-1.rst:2]:
```suggestion
  the `skipn_skipn` lemma
```

**Comment by stepchowfun** [comment]
Thanks @Alizter for the quick feedback! I see that Windows builds are [failing](https://github.com/coq/coq/actions/runs/3220497150/jobs/5267272989), but the error looks unrelated to this change. Is that expected?

File: theories/Lists/List.v
Comment (Alizter) [inline theories/Lists/List.v:2120]: Instead of using repeat, you can write `rewrite n skipn_nil` where n is the number of times you want to rewrite. Instead of reflexivity you could prefix the previous tactic with `now`.

File: theories/Lists/List.v
Comment (Alizter) [inline theories/Lists/List.v:2121]: `auto` here should be able to solve the first next goal given the right hint db. But @coq/stdlib-maintainers will know which one to use.

I think `destruct x, y` should also work.

**Comment by olaure01** [comment]
Two suggestions:
* the proof is not compatible with name mangling (`IHl` is used but not introduced for example)
* induction over `y` could make the proof slightly shorter
For instance:
```coq
  Proof.
    intros l x y; rewrite Nat.add_comm; induction y as [|y IHy] in l |- *.
    - reflexivity.
    - destruct l.
      + rewrite skipn_nil. reflexivity.
      + rewrite skipn_cons, IHy. reflexivity.
  Qed.
```

**Review by Alizter — COMMENTED** [review]
Some more comments

**Comment by Alizter** [comment]
@olaure01's proof is cleaner, just take what you will from my comments before and apply it to that.

File: theories/Lists/List.v
Comment (JasonGross) [reply theories/Lists/List.v:2120]: Note that `rewrite ?skipn_nil` is the same as `repeat rewrite skipn_nil`, and `rewrite !skipn_nil` is the same as `progress repeat rewrite skipn_nil`

**Comment by stepchowfun** [comment]
Thanks all for the feedback! I've tried to incorporate it where possible.

File: theories/Lists/List.v
Comment (stepchowfun) [reply theories/Lists/List.v]: 👍

File: theories/Lists/List.v
Comment (olaure01) [inline theories/Lists/List.v:2121]:
```suggestion
    - intros [|].
```
for compatibility with name mangling.

**Review by olaure01 — APPROVED** [review]

**Comment by olaure01** [comment]
@coqbot run full CI

**Comment by olaure01** [comment]
@stepchowfun CI failure looks unrelated
You can squash commits and I will merge by tomorrow if there is no more comments.

**Comment by stepchowfun** [comment]
Done—thank you!

**Comment by olaure01** [comment]
@coqbot merge now

**Pull request merged**
